diff -u -r vksdk_library.orig/build.gradle vksdk_library/build.gradle
--- build.gradle.orig	2018-07-02 15:42:17.272984573 +0300
+++ build.gradle	2018-07-02 15:42:06.780984702 +0300
@@ -1,17 +1,11 @@
 apply plugin: 'com.android.library'
 
 android {
-    compileSdkVersion 25
-    buildToolsVersion '25.0.2'
-
-    defaultConfig {
-        minSdkVersion 9
-        targetSdkVersion 25
-    }
+    compileSdkVersion androidCompileSdkVersion.toInteger()
+    buildToolsVersion androidBuildToolsVersion
 }
 
 configurations {
-    // Used for correct javadoc generation
     javadocDeps
 }
 
@@ -19,5 +13,3 @@
     javadocDeps 'com.android.support:support-annotations:25.1.0'
     compile 'com.android.support:support-v4:25.1.0'
 }
-
-apply from: '../maven_push.gradle'
\ No newline at end of file
Only in vksdk_library.orig: .gitignore
diff -u -r vksdk_library.orig/src/main/java/com/vk/sdk/api/VKBatchRequest.java vksdk_library/src/main/java/com/vk/sdk/api/VKBatchRequest.java
--- src/main/java/com/vk/sdk/api/VKBatchRequest.java.orig	2018-07-02 15:42:17.275984573 +0300
+++ src/main/java/com/vk/sdk/api/VKBatchRequest.java	2018-07-02 15:42:06.782984702 +0300
@@ -35,6 +35,7 @@
     private final VKResponse[] mResponses;
     private final VKRequest.VKRequestListener[] mOriginalListeners;
     private boolean mCanceled = false;
+    private boolean mErrorOccured = false;
 
     /**
      * Specify listener for current request
@@ -129,8 +130,8 @@
     }
 
     protected void provideError(VKError error) {
-        if (mCanceled)
-            return;
+        if (mErrorOccured) return;
+        mErrorOccured = true;
         for (int i = 0; i < mRequests.length; i++) {
             VKRequest.VKRequestListener l = mOriginalListeners[i];
             if (l != null) {
diff -u -r vksdk_library.orig/src/main/java/com/vk/sdk/VKAccessToken.java vksdk_library/src/main/java/com/vk/sdk/VKAccessToken.java
--- src/main/java/com/vk/sdk/VKAccessToken.java.orig	2018-07-02 15:42:17.273984573 +0300
+++ src/main/java/com/vk/sdk/VKAccessToken.java	2018-07-02 15:42:06.780984702 +0300
@@ -161,7 +161,7 @@
      *
      * @return Serialized token string as query-string
      */
-    protected String serialize() {
+    public String serialize() {
         return VKStringJoiner.joinParams(tokenParams());
     }
 
@@ -284,7 +284,7 @@
      * @param newToken New access token to set. If null, removes old token from preferences
      * @return old value of access token
      */
-    static VKAccessToken replaceToken(@NonNull Context ctx, @Nullable VKAccessToken newToken) {
+    public static VKAccessToken replaceToken(@NonNull Context ctx, @Nullable VKAccessToken newToken) {
         VKAccessToken oldToken = sCurrentToken;
         sCurrentToken = newToken;
         if (sCurrentToken != null) {
Only in vksdk_library.orig/src/main: project.properties
Only in vksdk_library.orig/src/main: .settings
